liquidator/liquidator.js

```diff
@@ -249,50 +269,129 @@ class Liquidator {
       );
 
       // Simple version of inventory management: simulate the transaction and assume that if it fails, the caller didn't have enough collateral.
-      let liquidationResult, gasEstimation;
-      try {
-        [liquidationResult, gasEstimation] = await Promise.all([
-          liquidation.call({ from: this.account }),
-          liquidation.estimateGas({ from: this.account })
+      const notEnoughCollateral = await liquidation
+        .call({ from: this.account })
+        .then(() => false)
+        .catch(() => true);
+
+      // Attempts to swap some capital for tokenCurrency
+      // Mutates the state of notEnoughCollateral in this logical scope
+      if (notEnoughCollateral) {
+        const tokenCurrency = await this.empClient.getTokenCurrency();
```

Comment:
Ideally, we don't want to have to run this async call here at all. We _already_ know this address from the `index.js`. If you need the `tokenCurrency` address consider adding it to the `empProps` so you can reduce the number of promise calls you to need to perform (especially for ones already made).

Reply:
Yeah why don't you pass it in as `tokenContract` similar to how `votingContract` is passed in?